File: LeetCode/cpp/1381_Design_a_Stack_With_Increment_Operation.cpp
class CustomStack {
public:
	stack<int> st;
	int maxNum = 0;

	CustomStack(int maxSize) {
		maxNum = maxSize;
	}

	void push(int x) {
		if (st.size() < maxNum) {
			st.push(x);
		}
	}

	int pop() {
		if (!st.empty()) {
			int n = st.top();
			st.pop();
			return n;
		}
		else {
			return -1;
		}
	}

	void increment(int k, int val) {
		vector<int> arr(st.size(), 0);
		int idx = st.size() - 1;

		while (!st.empty()) {
			arr[idx--] = st.top();
			st.pop();
		}

		for (int i = 0; i < k; i++) {
			if (i == arr.size()) {
				break;
			}

			arr[i] = arr[i] + val;
		}

		for (int i = 0; i < arr.size(); i++) {
			st.push(arr[i]);
		}
	}
};

/**
 * Your CustomStack object will be instantiated and called as such:
 * CustomStack* obj = new CustomStack(maxSize);
 * obj->push(x);
 * int param_2 = obj->pop();
 * obj->increment(k,val);
 */

File: LeetCode/java/1381_Design_a_Stack_With_Increment_Operation.java
import java.util.Stack;

class CustomStack {
    Stack<Integer> stack;
    int max = 0;

    public CustomStack(int maxSize) {
        stack = new Stack<>();
        max = maxSize;
    }

    public void push(int x) {
        if (stack.size() < max) {
            stack.push(x);
        }
    }

    public int pop() {
        if (!stack.empty()) {
            return stack.pop();
        } else {
            return -1;
        }
    }

    public void increment(int k, int val) {
        int[] arr = new int[stack.size()];
        int idx = stack.size() - 1;

        while (!stack.empty()) {
            arr[idx--] = stack.pop();
        }

        for (int i = 0; i < k; i++) {
            if (i == arr.length) {
                break;
            }

            arr[i] = arr[i] + val;
        }

        for (int i = 0; i < arr.length; i++) {
            stack.push(arr[i]);
        }
    }
}

/**
 * Your CustomStack object will be instantiated and called as such:
 * CustomStack obj = new CustomStack(maxSize);
 * obj.push(x);
 * int param_2 = obj.pop();
 * obj.increment(k,val);
 */